src/lib/nftClaimer/utils.ts

```diff
@@ -0,0 +1,131 @@
+import { gql, ApolloClient, InMemoryCache, HttpLink } from '@apollo/client/core';
+import fetch from 'cross-fetch';
+import snapshot from '@snapshot-labs/snapshot.js';
+import { CID } from 'multiformats/cid';
+import { Wallet } from '@ethersproject/wallet';
+import { getAddress, isAddress } from '@ethersproject/address';
+import { BigNumber } from '@ethersproject/bignumber';
+import type { Proposal, Space } from '../../helpers/snapshot';
+
+const requiredEnvKeys = [
+  'NFT_CLAIMER_PRIVATE_KEY',
+  'NFT_CLAIMER_NETWORK',
+  'NFT_CLAIMER_DEPLOY_VERIFYING_CONTRACT',
+  'NFT_CLAIMER_DEPLOY_IMPLEMENTATION_ADDRESS',
+  'NFT_CLAIMER_DEPLOY_INITIALIZE_SELECTOR',
+  'NFT_CLAIMER_SUBGRAPH_URL'
+];
+
+const HUB_NETWORK = process.env.NETWORK || '1';
+
+const missingEnvKeys: string[] = [];
+requiredEnvKeys.forEach(key => {
+  if (!process.env[key]) {
+    missingEnvKeys.push(key);
+  }
+});
+
+if (missingEnvKeys.length > 0) {
+  throw new Error(
+    `NFT Claimer not configured properly, missing env keys: ${missingEnvKeys.join(', ')}`
+  );
+}
+
+export const signer = new Wallet(process.env.NFT_CLAIMER_PRIVATE_KEY as string);
+
+async function mintingAllowed(space: Space) {
+  return (await getSpaceCollection(space.id)).enabled;
+}
+
+export async function validateSpace(address: string, space: Space | null) {
+  if (!space) {
+    throw new Error('RECORD_NOT_FOUND');
+  }
+
+  if (
+    (process.env.NFT_CLAIMER_NETWORK !== '5' || process.env.NODE_ENV === 'test') &&
+    (await snapshot.utils.getSpaceController(space.id, HUB_NETWORK)) !== getAddress(address)
+  ) {
+    throw new Error('Address is not the space owner');
+  }
+
+  const contract = await getSpaceCollection(space.id);
+  if (contract) {
+    throw new Error(`SpaceCollection contract already exist (${contract.id})`);
+  }
+}
+
+export function validateProposal(proposal: Proposal | null, proposer: string) {
+  if (!proposal) {
+    throw new Error('RECORD_NOT_FOUND');
+  }
+
+  if (getAddress(proposer) !== getAddress(proposal.author as string)) {
+    throw new Error('Proposal author is not matching');
+  }
+
+  if (!mintingAllowed(proposal.space)) {
+    throw new Error('Space has not allowed minting');
+  }
+}
+
+export async function getProposalContract(spaceId: string) {
+  const contract = await getSpaceCollection(spaceId);
+
+  if (!contract) {
+    throw new Error(`SpaceCollection contract is not found for space ${spaceId}`);
+  }
+
+  return contract.id;
+}
+
+const client = new ApolloClient({
```

Comment:
client is more simple, hard to find which one, Can we rename something like `nftClaimerClient`? anything of your choice,

Reply:
Why should we have a unique name ? This variable is local only, and never exported.

Reply:
Just felt I couldn't recognize it when i first reviewed it but should be fine if we leave it as it is